luasteam.cxx

```diff
@@ -3,109 +3,210 @@
 #include <string>
 
 #ifdef _WIN32
-	#define EXTERN extern "C" __declspec( dllexport )
+#    define EXTERN extern "C" __declspec(dllexport)
 #else
-	#define EXTERN extern "C"
+#    define EXTERN extern "C"
 #endif
 
 extern "C" {
-#include <lua.h>
 #include <lauxlib.h>
+#include <lua.h>
 }
 
-//GENERAL FUNCTIONS
-EXTERN int luasteam_init(lua_State *L){
-    lua_settop(L,0); //Reset lua stack
-    lua_pushboolean(L,SteamAPI_Init());
-    return 1;
-}
+namespace {
 
-EXTERN int luasteam_shutdown(lua_State *L){
-    SteamAPI_Shutdown();
-    return 0;
-}
+class SteamFriendsListener;
+
+lua_State *L = nullptr;
+int friends_ref = LUA_NOREF;
+SteamFriendsListener *friends_listener = nullptr;
+
+} // namespace
 
-//ACHIEVEMENTS FUNCTIONS
+// ==============================
+// ======= SteamUserStats =======
+// ==============================
 
-//bool GetUserAchievement(const char *pchName, bool *pbAchieved );
-EXTERN int luasteam_getAchievement(lua_State *L){
-    lua_settop(L,1);
-    if (!lua_isstring(L,1)){
+// bool GetUserAchievement(const char *pchName, bool *pbAchieved );
+EXTERN int luasteam_getAchievement(lua_State *L) {
+    lua_settop(L, 1);
+    if (!lua_isstring(L, 1)) {
         lua_pushstring(L, "incorrect type for argument1 (string expected)");
         return lua_error(L);
     }
-    std::string ach_name = lua_tostring(L,1);
-    lua_settop(L,0);
+    std::string ach_name = lua_tostring(L, 1);
+    lua_settop(L, 0);
     bool achieved = false;
-    bool success = SteamUserStats()->GetAchievement(ach_name.c_str(),&achieved);
-    lua_pushboolean(L,success);
-    lua_pushboolean(L,achieved);
+    bool success = SteamUserStats()->GetAchievement(ach_name.c_str(), &achieved);
```

Comment:
Creating a `std::string` to just send the `char*` forward is wasteful. If needed, just remove the `lua_settop`